core/src/main/java/nl/nn/adapterframework/webcontrol/api/ApiExceptionHandler.java

```diff
@@ -38,26 +38,25 @@
  */
 
 @Provider
-public class ApiExceptionHandler implements ExceptionMapper<Exception> {
+public class ApiExceptionHandler implements ExceptionMapper<WebApplicationException> {
 
 	private Logger log = LogUtil.getLogger(this);
 
 	@Override
-	@Produces(MediaType.TEXT_PLAIN)
-	public Response toResponse(Exception exception) {
+	public Response toResponse(WebApplicationException exception) {
 		//If the message has already been wrapped in an exception we don't need to `convert` it!
 		if(exception instanceof ApiException) {
 			return ((ApiException) exception).getResponse();
 		}
 
-		log.error("Caught exception in handling FF!API call", exception);
+		log.warn("Caught unhandled exception in handling FF!API call", exception);
 
-		ResponseBuilder response = Response.status(Status.INTERNAL_SERVER_ERROR);
+		ResponseBuilder response = Response.status(Status.INTERNAL_SERVER_ERROR).type(MediaType.TEXT_PLAIN);
 		String message = exception.getMessage();
 
 		if(message != null) {
 			message = message.replace("\"", "\\\"").replace("\n", " ").replace(System.getProperty("line.separator"), " ");
-			Map<String, Object> entity = new HashMap<String, Object>(3);
+			Map<String, Object> entity = new HashMap<>(3);
 			entity.put("status", "error");
 			entity.put("error", message);
 			entity.put("stackTrace", exception.getStackTrace());
```

Comment:
Het zat er al in, maar pen-testers slaan aan op stacktraces in error messages. Ik denk dat het goed zou zijn om die met een property in PRD uit te kunnen zetten.